src/main/java/io/cdap/plugin/aws/s3/common/S3ConnectorConfig.java

```diff
@@ -105,8 +116,9 @@ public boolean isAccessCredentials() {
     return ACCESS_CREDENTIALS.equalsIgnoreCase(authenticationMethod);
   }
 
-  public void validate(FailureCollector collector) {
-    if (containsMacro(NAME_AUTH_METHOD)) {
+  public void validate(FailureCollector collector, Boolean verifyCredentials) {
+    if (containsMacro(NAME_AUTH_METHOD) || containsMacro(NAME_ACCESS_ID) || containsMacro(NAME_ACCESS_KEY) ||
+      containsMacro(NAME_SESSION_TOKEN)) {
```

Comment:
This does not look correct, there can be a situation where `accessId` is `null` but `session_token` is macro.

This will lead to skipping some validations which should not be skipped.

Reply:
There can be a lot of combinations for these macro checks of all 3 credentials which can be added in the subsequent PRs. 
These issues were identified as legacy bugs in the QA verification activity and have been added to the backlog.

Reply:
Macro checks should not skip null checks.

This method can be easily refactored to something like:

```
public void validate(FailureCollector collector, Boolean verifyCredentials) {
    if (containsMacro(NAME_AUTH_METHOD)) {
         return;
    }
    if (!containsMacro(NAME_ACCESS_ID) && (accessID == null || accessID.isEmpty())) {
      collector.addFailure("The Access ID must be specified if authentication method is Access Credentials.", null)
        .withConfigProperty(NAME_ACCESS_ID).withConfigProperty(NAME_AUTH_METHOD);
      return;
    }
    if (!containsMacro(NAME_ACCESS_KEY) && (accessKey == null || accessKey.isEmpty())) {
      collector.addFailure("The Access Key must be specified if authentication method is Access Credentials.", null)
        .withConfigProperty(NAME_ACCESS_KEY).withConfigProperty(NAME_AUTH_METHOD);
      return;
    }
    if (verifyCredentials && !containsMacro(NAME_ACCESS_ID) && !containsMacro(NAME_ACCESS_KEY) && !containsMacro(NAME_SESSION_TOKEN)) {
      validateCredentials(collector);
    }
}

```

Reply:
Refactored. 78c37bf07fc8ed6e479d2f76a8d26c6dca218728
Thanks!